bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/asset/FileOrRendition.java

```diff
@@ -0,0 +1,158 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.impl.processes.asset;
+
+import com.adobe.acs.commons.mcp.impl.processes.asset.AssetIngestor.HierarchialElement;
+import com.adobe.acs.commons.mcp.impl.processes.asset.AssetIngestor.Source;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Abstraction of a file which might be an asset or a rendition of another asset
+ */
+public class FileOrRendition extends HashMap<String, String> implements HierarchialElement {
+    private final String url;
+    private final String name;
+    private final Folder folder;
+    private Source fileSource;
+    private final Map<String, FileOrRendition> additionalRenditions;
+    private String renditionName = null;
+    private String originalAssetName = null;
+    
+    public FileOrRendition(String name, String url, Folder folder, Map<String, String> data) {
+        if (folder == null) {
+            throw new NullPointerException("Folder cannot be null");
+        }
+        this.name = name;
+        this.url = url;
+        this.folder = folder;
+        putAll(data);
+        additionalRenditions = new TreeMap<>();
+    }
+    
+    public void setAsRenditionOfImage(String renditonName, String originalAssetName) {
+        this.renditionName = renditonName;
+        this.originalAssetName = originalAssetName;
+    }
+    
+    @Override
+    public boolean isFile() {
+        return true;
+    }
+
+    @Override
+    public boolean isFolder() {
+        return false;
+    }
+    
+    public boolean isRendition() {
+        return renditionName != null && !renditionName.isEmpty();
+    }
+    
+    public void addRendition(FileOrRendition rendition) {
+        additionalRenditions.put(rendition.renditionName, rendition);
+    }
+    
+    public Map<String, FileOrRendition> getRenditions() {
+        return additionalRenditions;
+    }
+    
+    public String getOriginalAssetName() {
+        return originalAssetName;
+    }
+
+    @Override
+    public HierarchialElement getParent() {
+        return folder;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getItemName() {
+        return getName();
+    }
+
+    @Override
+    public Source getSource() {
+        if (fileSource == null) {
+            FileOrRendition thizz = this;
+            fileSource = new Source() {
+                Long size = null;
+                URLConnection connection = null;
```

Comment:
*important* please use HttpClient here, not URLConnection, so that the proxy configuration from AEM is leveraged. Also, there should be configurable timeouts set (both connect and socket).

Reply:
Also, I suspect we need a close() method on the Source interface to ensure that the connection gets cleaned up when the source is not needed anymore.

Reply:
Actually URLConnection is kinda desired in case it is not an HTTP URL.  I'll add a switch block in there to use HTTP Client for cases where it is an HTTP Url, and URLConnection otherwise (for file:// or ftp://)

Reply:
That raises some more questions :) how would proxy support work for ftp? just document as a known issue.

What about timeouts for FTP?